Add Math Books from aimath.org

SethFalco opened the pull request:
## What does this PR do?
Add resource(s) | Improve repo

## For resources
### Description 
* Part of #5206 (Does not fix/complete the entire issue.)
  * Adds almost every single book except the ones under "Books Using Sage"
  * Some books that already existed in the repo, I updated to include authors, or a direct link to the PDF.
  * Created Language Agnostic / Math section in French, German, Spanish, and Korean to include translated resources.

This does not link directly to the resources from aimath.org, but rather goes into the links they link to. (They're just sharing links to external sites.)  
I've checked each one, and downloading the PDF to confirm they're cool.

<details>
  <summary>I made this as a separate list first, then inserted it to the repo after. Keeping the list here here for convenience. (ie if this is rejected, or should be moved to free-science-books or something instead.)</summary>

```md
* [Math in Society](https://www.opentextbookstore.com/mathinsociety/) - David Lippman [(PDF)](https://www.opentextbookstore.com/mathinsociety/current2.php?chapter=MathinSociety.pdf)
* [Mathematical Discovery](https://classicalrealanalysis.info/com/Mathematical-Discovery.php) - Andrew M. Bruckner, Brian S. Thomson, and Judith B. Bruckner [(PDF)](http://classicalrealanalysis.info/com/documents/BTB-colorA.pdf)
* [ORCCA: Open Resources for Community College Algebra](https://spaces.pcc.edu/pages/viewpage.action?pageId=52729944) - Portland Community College [(PDF)](https://spot.pcc.edu/math/orcca/ed2/pdf/orcca-screenpdf.pdf)
* [Precalculus / College Algebra / Trigonometry](https://stitz-zeager.com/) - Carl Stitz and Jeff Zeager [(PDF)](https://stitz-zeager.com/szprecalculus07042013.pdf)
* [Active Calculus](https://scholarworks.gvsu.edu/books/20/) - Matt Boelkins [(PDF)](https://scholarworks.gvsu.edu/cgi/viewcontent.cgi?article=1020&context=books)
* [APEX Calculus](https://www.apexcalculus.com/) - Gregory Hartman, Brian Heinold, Troy Siemers, and Dimplekumar Chalishajar [(PDF)](https://drive.google.com/open?id=1bDsqf4u3rNz_dVnlViE_AXrhWpJoBi3H)
* [CLP-1 Differential Calculus](https://www.math.ubc.ca/~CLP/CLP1/) - Joel Feldman, Andrew Rechnitzer, and Elyse Yeager [(PDF)](https://www.math.ubc.ca/~CLP/CLP1/clp_1_dc_text.pdf) 
* [CLP-2 Integral Calculus](https://www.math.ubc.ca/~CLP/CLP2/) - Joel Feldman, Andrew Rechnitzer, and Elyse Yeager [(PDF)](https://www.math.ubc.ca/~CLP/CLP2/clp_2_ic_text.pdf) 
* [CLP-3 Multivariable Calculus](https://www.math.ubc.ca/~CLP/CLP3/) - Joel Feldman, Andrew Rechnitzer, and Elyse Yeager [(PDF)](https://www.math.ubc.ca/~CLP/CLP3/clp_3_mc_text.pdf) 
* [CLP-4 Vector Calculus](https://www.math.ubc.ca/~CLP/CLP4/) - Joel Feldman, Andrew Rechnitzer, and Elyse Yeager [(PDF)](https://www.math.ubc.ca/~CLP/CLP4/clp_4_vc_text.pdf) 
* [Calculus in Context](http://www.math.smith.edu/~callahan/intromine.html) - James Callahan [(PDF)](http://www.math.smith.edu/~callahan/cic/book.pdf)
* [Calculus I](https://resolver.caltech.edu/CaltechBOOK:1985.001) - Jerrold E. Marsden and Alan Weinstein [(PDF)](https://authors.library.caltech.edu/25030/1/Calc1w.pdf)
* [Calculus](https://ocw.mit.edu/resources/res-18-001-calculus-online-textbook-spring-2005/textbook/) - Gilbert Strang [(PDF)](http://ocw.mit.edu/ans7870/resources/Strang/Edited/Calculus/Calculus.pdf)
* [Vector Calculus](http://www.mecmath.net/) - Michael Corral [(PDF)](http://www.mecmath.net/VectorCalculus.pdf)
* [Notes on Diffy Qs](https://www.jirka.org/diffyqs/) - Jiří Lebl [(PDF)](https://www.jirka.org/diffyqs/diffyqs.pdf)
* [Elementary Differential Equations (with Boundary Value Problems)](https://digitalcommons.trinity.edu/mono/9/) - William F. Trench [(PDF)](https://digitalcommons.trinity.edu/cgi/viewcontent.cgi?article=1008&context=mono)
* [A First Course in Linear Algebra](http://linear.ups.edu/) - Rob Beezer [(PDF)](http://linear.ups.edu/download/fcla-3.50-tablet.pdf)
* [Linear Algebra](http://joshua.smcvt.edu/linearalgebra/) - Jim Hefferon [(PDF)](http://joshua.smcvt.edu/linearalgebra/book.pdf)
* [Linear Algebra Done Wrong](https://www.math.brown.edu/streil/papers/LADW/LADW.html) - Sergei Treil [(PDF)](https://www.math.brown.edu/streil/papers/LADW/LADW_2017-09-04.pdf)
* [Book of Proof](https://www.people.vcu.edu/~rhammack/BookOfProof/) - Richard Hammack [(PDF)](https://www.people.vcu.edu/~rhammack/BookOfProof/Main.pdf)
* [A Gentle Introduction to the Art of Mathematics](https://osj1961.github.io/giam/) - Joseph E. Fields [(PDF)](https://github.com/osj1961/giam/blob/master/GIAM.pdf?raw=true)
* [Mathematical Reasoning: Writing and Proof](https://www.tedsundstrom.com/mathematical-reasoning-3) - Ted Sundstrom [(PDF)](https://scholarworks.gvsu.edu/cgi/viewcontent.cgi?article=1024&context=books)
* [Applied Discrete Structures](https://faculty.uml.edu/klevasseur/ads2/) - Alan Doerr and Kenneth Levasseur [(PDF)](http://discretemath.org/ads-latex/ads.pdf)
* [Discrete Mathematics: An Open Introduction](http://discrete.openmathbooks.org/dmoi3.html) - Oscar Levin [(PDF)](http://discrete.openmathbooks.org/pdfs/dmoi-tablet.pdf)
* [Discrete Mathematics: First and Second Course](https://cseweb.ucsd.edu/~gill/BWLectSite/) - Edward A. Bender and S. Gill Williamson
* [Applied Combinatorics](https://rellek.net/book/app-comb.html) - Mitchel T. Keller and William T. Trotter
* [Combinatorics Through Guided Discovery](https://bogart.openmathbooks.org/) - Kenneth Bogart [(PDF)](https://bogart.openmathbooks.org/pdf/ctgd.pdf)
* [Foundations of Combinatorics with Applications](https://www.math.ucsd.edu/~ebender/CombText/) - Edward A. Bender and S. Gill Williamson (PDF)
* [Sage for Undergraduates](http://www.gregorybard.com/books.html) - Gregory Bard [(PDF)](http://www.gregorybard.com/sage_for_undergraduates_color.pdf.zip) + SPANISH http://www.sage-para-estudiantes.com/
* [Computational Mathematics with SageMath](http://sagebook.gforge.inria.fr/english.html) - Paul Zimmermann et al. [(PDF)](http://dl.lateralis.org/public/sagebook/sagebook-ba6596d.pdf) + FRENCH AND GERMAN
* [Tea Time Numerical Analysis](https://lqbrin.github.io/tea-time-numerical/) - Leon Q. Brin [(PDF)](https://github.com/lqbrin/tea-time-numerical/blob/master/TeaTimeNumericalAnalysis.pdf?raw=true)
* [Number Theory: In Context and Interactive](http://math.gordon.edu/ntic/) - Karl-Dieter Crisman [(PDF)](http://math.gordon.edu/ntic/ntic.pdf)
* [Elementary Number Theory: Primes, Congruences, and Secrets](https://wstein.org/ent/) - William Stein [(PDF)](https://wstein.org/ent/ent.pdf) + KOREAN
* [A Computational Introduction to Number Theory and Algebra](https://shoup.net/ntb/) - Victor Shoup [(PDF)](https://shoup.net/ntb/ntb-v2.pdf)
* [Abstract Algebra: Theory and Applications](http://abstract.ups.edu/) - Tom Judson [(PDF)](http://abstract.ups.edu/download/aata-20200730.pdf)
* [Elements of Abstract and Linear Algebra](https://www.math.miami.edu/~ec/book/) - E. H. Connell [(PDF)](https://www.math.miami.edu/~ec/book/book.pdf)
* [Algebra: Abstract and Concrete](https://homepage.divms.uiowa.edu/~goodman/algebrabook.dir/algebrabook.html) - Frederick Goodman [(PDF)](https://homepage.divms.uiowa.edu/~goodman/algebrabook.dir/book.2.6.pdf)
* [Basic Analysis: Introduction to Real Analysis](https://www.jirka.org/ra/) - Jiří Lebl [(PDF)](https://www.jirka.org/ra/realanal.pdf)
* [Introduction to Real Analysis](https://digitalcommons.trinity.edu/mono/7/) - William F. Trench [(PDF)](https://digitalcommons.trinity.edu/cgi/viewcontent.cgi?article=1006&context=mono)
* [Elementary Real Analysis](https://www.classicalrealanalysis.info/com/Elementary-Real-Analysis.php) - Brian S. Thomson, Judith B. Bruckner, Andrew M. Bruckner [(PDF)](http://classicalrealanalysis.info/com/documents/TBB-AllChapters-Portrait.pdf)
* [Mathematical Analysis I](http://www.trillia.com/zakon-analysisI.html) - Elias Zakon [(PDF)](http://www.trillia.com/dA/zakon-analysisI-a4-one.pdf)
* [How We Got from There to Here: A Story of Real Analysis](https://milneopentextbooks.org/how-we-got-from-there-to-here-a-story-of-real-analysis/) - Robert Rogers and Eugene Boman [(PDF)](https://milneopentextbooks.org/download/asora_ebook-pdf/)
* [A First Course in Complex Analysis](http://math.sfsu.edu/beck/complex.html) - Matthias Beck, Gerald Marchesi, Dennis Pixton, Lucas Sabalka [(PDF)](http://math.sfsu.edu/beck/papers/complex.pdf)
* [Geometry with an Introduction to Cosmic Topology](https://mphitchman.com/) - Michael P. Hitchman [(PDF)](https://mphitchman.github.io/geometry/GCTscreen.pdf)
* [Introduction to Probability](https://www.dartmouth.edu/~chance/teaching_aids/books_articles/probability_book/book.html) - Charles M. Grinstead and J. Laurie Snell [(PDF)](https://www.dartmouth.edu/~chance/teaching_aids/books_articles/probability_book/amsbook.mac.pdf)
* [Probability: Lectures and Labs](https://www.markhuberdatascience.org/probability-textbook) - Mark Huber [(PDF)](https://d71b37a6-f420-4d99-8d4f-f64245d5a10b.filesusr.com/ugd/c2b9b6_2738b4945c4042b9b9e4e22468947c4f.pdf)
* [OpenIntro Statistics](https://www.openintro.org/book/stat/) - David M. Diez, Christopher D. Barr, Mine Çetinkaya-Rundel
* [Introductory Statistics for the Life and Biomedical Sciences](https://www.openintro.org/book/isrs/) - Julie Vu and David Harrington [(PDF)](https://leanpub.com/ims)
* [SticiGui](https://www.stat.berkeley.edu/~stark/SticiGui/) - Philip Stark
* [Online Statistics Education](http://onlinestatbook.com/) - David Lane, lead author [(PDF)](http://onlinestatbook.com/Online_Statistics_Education.pdf)
* [A Friendly Introduction to Mathematical Logic](https://milneopentextbooks.org/a-friendly-introduction-to-mathematical-logic/) - Christopher C. Leary and Lars Kristiansen [(PDF)](https://milneopentextbooks.org/download/friendly-intro-2e-update19-pdf/)
```
</details>

### Why is this valuable (or not)?
Gave some book a small peak, seem really useful for advance math topics.

### How do we know it's really free?
Most of them are released under Creative Commons, and all were linked from aimath.org as Open Text Books.

### For book lists, is it a book? For course lists, is it a course? etc.
Books

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists  in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Followup

- Check the output of Travis-CI for linter errors!

eshellman commented:
Is there a reason to separately include the pdf link when the pdf is available from the primary link? many of them are versioned and thus more likely to break than the main links, and thus adding to our maintenance burden.

eshellman commented:
Also, Would like to see opinions on #5249. I'm not excited about adding a category consisting of one subcategory with one item. The point of grouping topics together is to make them easier to browse, and for purposes of this PR, I think creating just a "Mathematics" category does this, assuming that there are no other mathematics books on a page. Also, note that using english language categories for German, French, Spanish and Korean pages is not optimal - I think most readers of these languages understand "mathematics", but "language agnostic"?

SethFalco commented:
@eshellman

> Is there a reason to separately include the pdf link when the pdf is available from the primary link?

It was just for convenience since I was grabbing the PDF to download for myself anyway.
I get what you mean though, I can remove them, or at least the versioned ones if that's preferred?

> I'm not excited about adding a category consisting of one subcategory with one item. The point of grouping topics together is to make them easier to browse, and for purposes of this PR, I think creating just a "Mathematics" category does this, assuming that there are no other mathematics books on a page.

Sure, I'll address this.

> Also, note that using english language categories for German, French, Spanish and Korean pages is not optimal - I think most readers of these languages understand "mathematics", but "language agnostic"?

Valid point, didn't stop to consider that tbh. :thinking:

SethFalco commented:
Fixed the language/category issues for now since they were quick.
I'll fix the PDF version links in another commit.

SethFalco commented:
* Fixed some poor formatting. (Accidentally removed a line break between two entries before.)
* Removed all PDF links I added, except in cases where the link was already in the repo before this contribution.

eshellman commented on books/free-programming-books-de.md:
```diff
@@ -149,6 +150,11 @@
 
+### Mathematik
+
+* [Calcul mathématique avec SAGE](http://sagebook.gforge.inria.fr/english.html) - Paul Zimmermann et al.
```
are you sure this is in german?

SethFalco replied:
The page is in English, but at the bottom there is a link to the German version.

> Traductions/Translations
>
> * German translation, contributed by Helmut Büch
> * French version (May 2013, 468 pages, for Sage 5.9)

Unfortunately there isn't a dedicated German page that hosts the download unlike with English and French.

Since for this it's just a PDF anyway, we could make the German one reference the link directly to skip the English/French page of that's preferred?

```md
[Title](http://www.loria.fr/~zimmerma/sagebook/CalculDeutsch.pdf) - Author (PDF)
```
